src/Locator/FileNotFound.php

```diff
@@ -63,8 +75,27 @@ public static function multipleFilesDoNotExist(string $path, array $files): self
         );
     }
 
-    public static function filesNotFound(): self
+    /**
+     * @param string[] $files
+     * @param string[] $roots
+     */
+    public static function createForFiles(array $files, array $roots): self
     {
-        return new self('Files are not found');
+        Assert::allString($files);
+        Assert::allString($roots);
+
+        if ([] === $files) {
+            $message = 'Could not locate any files (no file provided).';
```

Comment:
What do you think about using an early return here, so we can avoid the `else` branch, and perhaps make it more obvious that the latter is the happy path?

Reply:
We can actually completely get rid of the if statement with a ternary and the `$message` variable which is a single usage variable